apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -7,66 +7,47 @@ import clsx from 'clsx';
 import { useState, useCallback, useEffect } from 'react';
 import { type Direction } from 'react-resizable-panels';
 
-import ModuleView from './ModuleView';
 import { ModuleFunctionsInteraction } from './module-functions-interaction';
 import { useBreakpoint } from '~/hooks/useBreakpoint';
 import { SplitPanes } from '~/ui/SplitPanes';
 import { TabHeader } from '~/ui/Tabs';
 import { ListItem, VerticalList } from '~/ui/VerticalList';
 import { useSearchParamsMerged } from '~/ui/utils/LinkWithQuery';
+import { ModuleCodeTabs } from './ModuleCodeTabs';
 
 type ModuleType = [moduleName: string, code: string];
 
 interface Props {
-	id?: string;
+	id: string;
 	modules: ModuleType[];
 	splitPanelOrientation: Direction;
 }
 
-interface ModuleViewWrapperProps {
-	id?: string;
-	selectedModuleName: string;
-	modules: ModuleType[];
-}
-
-function ModuleViewWrapper({ id, selectedModuleName, modules }: ModuleViewWrapperProps) {
-	const selectedModuleData = modules.find(([name]) => name === selectedModuleName);
-
-	if (!selectedModuleData) {
-		return null;
-	}
-
-	const [name, code] = selectedModuleData;
-
-	return <ModuleView id={id} name={name} code={code} />;
-}
-
 function PkgModuleViewWrapper({ id, modules, splitPanelOrientation }: Props) {
 	const isMediumOrAbove = useBreakpoint('md');
 
-	const modulenames = modules.map(([name]) => name);
 	const [searchParams, setSearchParams] = useSearchParamsMerged();
 	const [query, setQuery] = useState('');
 
-	// Extract module in URL or default to first module in list
-	const selectedModule =
-		searchParams.get('module') && modulenames.includes(searchParams.get('module')!)
-			? searchParams.get('module')!
-			: modulenames[0];
+	const moduleNameValue = searchParams.get('module');
+	const moduleFromParams = moduleNameValue
```

Comment:
I refactored some of this for better type safety / easier readability / easier data plumbing